packages/mds-cache/unflatteners.ts

```diff
@@ -1,11 +1,162 @@
+          event_type_reason: tripsEvents[trip_id][i].event_type_reason
+            ? (tripsEvents[trip_id][i].event_type_reason as VEHICLE_REASON)
+            : null,
+          annotation_version: Number(tripsEvents[trip_id][i].annotation_version),
+          annotation: {
+            in_bound: JSON.parse(tripsEvents[trip_id][i].annotation.in_bound),
+            areas: tripsEvents[trip_id][i].annotation.areas
+          },
+          gps: {
+            lat: Number(tripsEvents[trip_id][i].gps.lat),
+            lng: Number(tripsEvents[trip_id][i].gps.lng),
+            altitude: tripsEvents[trip_id][i].gps.altitude ? Number(tripsEvents[trip_id][i].gps.altitude) : null,
+            heading: tripsEvents[trip_id][i].gps.heading ? Number(tripsEvents[trip_id][i].gps.heading) : null,
+            speed: tripsEvents[trip_id][i].gps.speed ? Number(tripsEvents[trip_id][i].gps.speed) : null,
+            accuracy: tripsEvents[trip_id][i].gps.accuracy ? Number(tripsEvents[trip_id][i].gps.accuracy) : null
+          },
+          service_area_id: tripsEvents[trip_id][i].service_area_id
+            ? (tripsEvents[trip_id][i].service_area_id as UUID)
+            : null
+        }
+      }
+    }
+    return trips
+  } catch (err) {
+    console.log(err)
+    throw new Error(`unable to parse tripsEvents: ${tripsEvents}`)
+  }
+}
+
+function parseTripsTelemetry(tripsTelemetry: StringifiedTripsTelemetry): TripsTelemetry {
+  try {
+    let trips: TripsTelemetry = {}
+    for (let trip_id in tripsTelemetry) {
+      for (let i = 0; i < tripsTelemetry[trip_id].length; i++) {
+        console.log('HERERERE1')
+        console.log(tripsTelemetry)
+        console.log('check')
+        console.log(tripsTelemetry[trip_id])
+        trips[trip_id][i] = {
+          timestamp: Number(tripsTelemetry[trip_id][i].timestamp) as Timestamp,
+          latitude: Number(tripsTelemetry[trip_id][i].latitude),
+          longitude: Number(tripsTelemetry[trip_id][i].longitude),
+          annotation_version: Number(tripsTelemetry[trip_id][i].annotation_version),
+          annotation: {
+            in_bound: JSON.parse(tripsTelemetry[trip_id][i].annotation.in_bound),
+            areas: tripsTelemetry[trip_id][i].annotation.areas
+          },
+          service_area_id: tripsTelemetry[trip_id][i].service_area_id
+            ? (tripsTelemetry[trip_id][i].service_area_id as UUID)
+            : null
+        }
+      }
+    }
+    return trips
+  } catch (err) {
+    console.log(err)
+    throw new Error(`unable to parse tripsTelemetry: ${tripsTelemetry}`)
+  }
+}
+
+function parseAllTripsEvents(allTripsEvents: StringifiedAllTripsEvents): { [vehicle_id: string]: TripsEvents } {
+  try {
+    let allTrips: { [vehicle_id: string]: TripsEvents } = {}
+    for (let vehicle_id in allTripsEvents) {
+      allTrips[vehicle_id] = parseTripsEvents(allTripsEvents[vehicle_id])
+    }
+    return allTrips
+  } catch (err) {
+    throw new Error(`unable to parse allTripsEvents`)
```

Comment:
I think it'd be nice to create an `Error` type just for parse errors like this, so in the future if we have a lot of issues with parsing data, it'll be easier to search through Kibana. See `mds-utils/exceptions.ts` for examples.